bot/exts/moderation/alts.py

```diff
@@ -0,0 +1,169 @@
+import gettext
+
+import discord
+from discord.ext import commands
+from pydis_core.site_api import ResponseCodeError
+from pydis_core.utils.members import get_or_fetch_member
+
+from bot import constants
+from bot.bot import Bot
+from bot.converters import UnambiguousMemberOrUser
+from bot.log import get_logger
+from bot.pagination import LinePaginator
+from bot.utils.channel import is_mod_channel
+from bot.utils.time import discord_timestamp
+
+log = get_logger(__name__)
+
+
+class AlternateAccounts(commands.Cog):
+    """A cog used to track a user's alternative accounts across Discord."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @staticmethod
+    def error_text_from_error(error: ResponseCodeError) -> str:
+        """Format the error into a user-facing message."""
+        if resp_json := error.response_json:
+            errors = ", ".join(
+                resp_json.get("non_field_errors", []) +
+                resp_json.get("source", []) +
+                resp_json.get("detail", [])
+            )
+            if errors:
+                return gettext.ngettext("Error from site: ", "Errors from site: ", len(errors)) + errors
+            return str(error.response_json)
+        return error.response_text
+
+    async def alts_to_string(self, alts: list[dict[str: str | int | list[int]]]) -> list[str]:
```

Comment:
I was going to leave it as `list[dict]` but thought I might as well expand it further. Didn't want to make a full model since site might change and I don't want ot have to maintain it in two places

Reply:
Why would you have to maintain it in 2 places ? 
Reading this as it is is "painful" and one would need to squeeze their toughts in order to understand what's going on, and it isn't pleasant. Giving it structure would be more readable/intuitive/understandable.

Reply:
I think that the current type isn't very useful ,`str | int | list[int]` makes the usages below technically incorrect since whenever accessing a value they should handle it being any of those types.

I would be find with any of these, since they're all "correct", it just depends what you consider in-scope for the PR given our typing elsewhere with infractions isn't great either afaik:
1) `list[dict[str, typing.Any]]`
2) Use a `TypedDict`
3) Use a Pydantic model

Reply:
Simplified the type annotation in https://github.com/python-discord/bot/pull/2845/commits/a380d2e030cf3f206e5e4d359db228e06cac81f5 (option 1)